File: backend/src/main/java/com/ieum/notice/domain/model/Notice.java
package com.ieum.notice.domain.model;

import lombok.*;

import java.time.LocalDateTime;

/**
 * 공지사항 도메인 모델 (순수 자바 객체)
 * - JPA 의존성 없음
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notice {

    private Long id;
    private String title;
    private String content;
    private String summary;

    @Builder.Default
    private NoticeCategory category = NoticeCategory.GENERAL;

    @Builder.Default
    private Integer viewCount = 0;

    @Builder.Default
    private Boolean isPinned = false;

    @Builder.Default
    private Boolean isPopup = false;

    @Builder.Default
    private Boolean isPushed = false;

    @Builder.Default
    private Boolean isActive = true;

    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/notice.ts
/**
 * 공지사항 타입 (백엔드 Notice 도메인 모델 매핑)
 */
export interface Notice {
  id: number;
  title: string;
  content: string;
  summary?: string;
  viewCount: number;
  isPinned: boolean;
  isPopup: boolean;
  isActive: boolean;
  startDate?: string;
  endDate?: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * 공지사항 상세 API 응답 (이전글/다음글 포함)
 */
export interface NoticeDetailResponse {
  notice: Notice;
  prevNotice?: { id: number; title: string } | null;
  nextNotice?: { id: number; title: string } | null;
}

/**
 * Spring Data Page 응답 구조
 */
export interface NoticePage {
  content: Notice[];
  totalElements: number;
  totalPages: number;
  number: number;         // 0-based page index
  size: number;
  first: boolean;
  last: boolean;
  empty: boolean;
}
